### PR TITLE
chore: init gen pkpass file

### PR DESCRIPTION
**Pull request by NguyenHuy1812**
**What does this PR do?**

-   [x] Add handle save apple wallet passes

https://github.com/consolelabs/mochi-discord/assets/28944972/50d05fef-4417-45a8-8dc1-c4bc86a59c12

### CODE REVIEW

File: src/commands/qr/index/processor.ts [hunk 1]
```diff
@@ -391,6 +392,47 @@ export async function viewQR(i: SelectMenuInteraction) {
...
+
+  return `${API_BASE_URL}/pk-pass?category=${encodeURIComponent(
+    category
+  )}&qr_value=${encodeURIComponent(qrValue)}&amount=${encodeURIComponent(
+    amount
+  )}&symbol=${encodeURIComponent(symbol)}&content=${encodeURIComponent(
+    content
+  )}&type=${encodeURIComponent(type)}`
+}
+
```
Comment (anhnh12) [inline src/commands/qr/index/processor.ts:435]: suggestion: use URL() to build url for more readable
```
const url = new URL(`${API_BASE_URL}/pk-pass`)
url.searchParams.append(......, .....)
url.searchParams.append(......, .....)
```

File: src/commands/qr/index/processor.ts [hunk 2]
```diff
@@ -184,10 +185,10 @@ async function compose(user: User, page: number) {
         rowAfterFormatter: (f, i) => {
           const d = data[i]
           const emoji = mapEmoji(d)
-
+          const url = d.category === "profile" ? HOMEPAGE_URL : buildQueryUrl(d)
```
Comment (anhnh12) [inline src/commands/qr/index/processor.ts:188]: it's good if we can have enum for category

**Review by trkhoi — APPROVED** [review]

File: src/commands/qr/index/processor.ts (hunk 2 above)
Comment (NguyenHuy1812) [reply src/commands/qr/index/processor.ts:188]: noted, will try it later, tks ser

File: src/commands/qr/index/processor.ts [hunk 3]
```diff
@@ -391,6 +392,50 @@ export async function viewQR(i: SelectMenuInteraction) {
...
+  }
+
+  url.searchParams.append("qr_value", qrValue)
+  url.searchParams.append("amount", amount)
+  url.searchParams.append("symbol", symbol)
+  url.searchParams.append("content", content)
+  url.searchParams.append("type", type)
+
+  return url.toString()
+}
```
Comment (anhnh12) [inline src/commands/qr/index/processor.ts:437]: can u revert back to the previous one? I remember a bug when using URL searchparams which may happen to `content`

```
return `${API_BASE_URL}/pk-pass?category=${encodeURIComponent(	  url.searchParams.append("qr_value", qrValue)
    category	  url.searchParams.append("amount", amount)
  )}&qr_value=${encodeURIComponent(qrValue)}&amount=${encodeURIComponent(	  url.searchParams.append("symbol", symbol)
    amount	  url.searchParams.append("content", content)
  )}&symbol=${encodeURIComponent(symbol)}&content=${encodeURIComponent(	  url.searchParams.append("type", type)
    content	
  )}&type=${encodeURIComponent(type)}`
```

**Review by anhnh12 — APPROVED** [review]

**Pull request merged**